app/bot/openai.go

```diff
@@ -26,6 +27,9 @@ type OpenAI struct {
 	prompt    string
 	superUser SuperUser
 
+	history LimitedMessageHistory
+	rand    func(n int64) int64 // tests may change it
```

Comment:
not sure if this even needed - you can achieve predefined rand sequence by setting fixed seed

Reply:
I just copied the approach by `wtf.go`